crates/pep508-rs/src/marker/tree.rs

```diff
@@ -1808,12 +1894,55 @@ mod test {
         assert_false("python_version == '3.9.0.*'");
         assert_true("python_version != '3.9.1'");
 
+        // Technically these is are valid substring comparisosn, but we do not allow them.
+        // e.g., using a version with patch components with `python_version` is considered
+        // impossible to satisfy since it is truncated at the minor version
+        assert_false("python_version in '3.9.0'");
+        // ... unless there are other versions that could match
+        assert_true("python_version in '3.* 3.9.0'");
```

Comment:
Why is this always `true`?

Reply:
Because `3.*` matches all known Python versions

Reply:
It could be false in the future though right?

Reply:
Does it? `python_version == '3.*'` simplifies to `python_full_version >= '3' and python_full_version < '4'` so I'm not sure why it would be different here. We aren't calling `simplify_python_versions` in this test case.

Reply:
Oh, I guess it's not actually a valid PEP 440 version (because it requires an operator to create the equal star).

Thanks — I'll update the comment accordingly. We might need to update the list to be `VersionSpecifier` instead of `Version` someday.

Reply:
Oh yeah sorry, I guess this would never be true since it's literally a string match.

Reply:
Yeah.. we but someone could do `python_version in '3.10.*'` and that should be true sometimes since it's a substring match — anyway clearly some caveats around this implementation I'll try to make the tests clearer.

Reply:
Lol true